Parser/Controllers/CrawlsController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using MongoDB.Driver;
+using System.Web.Http;
+using Visitors.Models;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+
+namespace Parser.Controllers
+{
+    public class CrawlsController : ApiController
+    {
+        private IMongoCollection<DomainMask> CollectionMasks { get; set; }
```

Comment:
Controllers should be just as dispatchers: they should not contain any business logic.
Create a separate project and move out the DB-related functionality there.
Converting of the database should be done not via Web API - create a "MsTest" project and create separate tests for converting different types of data.